lib/hooks/core/hook-http.js

```diff
@@ -35,6 +35,7 @@ function requestWrap(request) {
 
     var localOptions = (typeof options === 'string') ?
       url.parse(options) : options;
+    localOptions.headers = localOptions.headers || {};
```

Comment:
I've made things more rigorous! We only copy when we know we need to and we don't modify their options object at all.